packages/node_modules/@ciscospark/react-redux-spark/src/component.js

```diff
@@ -27,71 +30,116 @@ const defaultProps = {
   guestToken: ''
 };
 
-class SparkComponent extends Component {
-  static listenToSparkEvents(spark, props) {
-    spark.listenToAndRun(spark, 'change:canAuthorize', () => {
-      props.updateSparkStatus({authenticated: spark.canAuthorize});
-    });
+const PLUGINS = ['authorization', 'logger', 'people', 'phone', 'rooms'];
+const INTERNAL_PLUGINS = ['conversation', 'feature', 'flag', 'mercury', 'presence', 'search', 'team'];
 
-    spark.listenToAndRun(spark, 'change:isAuthenticating', () => {
-      props.updateSparkStatus({authenticating: spark.isAuthenticating});
-    });
+export class SparkComponent extends Component {
+  constructor(props) {
+    super(props);
 
-    spark.listenToAndRun(spark.internal.device, 'change:registered', () => {
-      props.updateSparkStatus({registered: spark.internal.device.registered});
-    });
-
-    SparkComponent.setupDevice(spark, props);
+    this.loadSparkInstance = this.loadSparkInstance.bind(this);
   }
 
   componentDidMount() {
-    const {props} = this;
     const {
       accessToken,
-      guestToken
+      guestToken,
+      spark
     } = this.props;
 
-    const spark = props.spark.get('spark');
+    const sparkInstance = spark.get('spark');
 
-    function storeAndListen(sparkInstance) {
-      props.storeSparkInstance(sparkInstance);
-      SparkComponent.listenToSparkEvents(sparkInstance, props);
-    }
-    if (!spark) {
+    if (!sparkInstance) {
       if (guestToken) {
-        createSparkJwt(guestToken, this.props).then(storeAndListen);
+        createSparkJwt(guestToken, this.props).then(this.loadSparkInstance);
       }
       else {
-        createSpark(accessToken, this.props).then(storeAndListen);
+        createSpark(accessToken, this.props).then(this.loadSparkInstance);
       }
     }
     else {
-      SparkComponent.listenToSparkEvents(spark, props);
+      this.listenToSparkEvents(sparkInstance);
     }
   }
 
-  componentWillReceiveProps(nextProps) {
-    const spark = nextProps.spark.get('spark');
-
-    SparkComponent.setupDevice(spark, nextProps);
-  }
-
-  shouldComponentUpdate(nextProps) {
-    const {props} = this;
-
-    return nextProps.spark !== props.spark;
-  }
-
-  static setupDevice(spark, props) {
+  /**
+   * Register the device if the user has been authenticated
+   * and the device is not regusterd yet.
+   * @param {Object} spark
```

Comment:
Let's update this to be `sparkInstance` too so we have consistency throughout the file.